### PR TITLE
Try to stablize FixingGhostMailboxTest

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

**Comment by Arsnael** [comment]
Did you try to repeat it like 50 times with `@RepeatedTest` to see if it breaks?

**Comment by Arsnael** [comment]
Don't hesitate to rebase and put back your commit change that I reverted in #3052

**Comment by trantienduchn** [comment]
```
[7aa172e6365d0379fefbc205e2e2d154437675f1] [ERROR] Errors: 
[7aa172e6365d0379fefbc205e2e2d154437675f1] [ERROR]   FixingGhostMailboxTest.ghostMailboxBugShouldChangeMailboxId » ParameterResolution
[7aa172e6365d0379fefbc205e2e2d154437675f1] [ERROR]   FixingGhostMailboxTest.ghostMailboxBugShouldDiscardOldContent » ParameterResolution
[7aa172e6365d0379fefbc205e2e2d154437675f1] [ERROR]   FixingGhostMailboxTest.mergingMailboxTaskShouldBeInformative » ParameterResolution
[7aa172e6365d0379fefbc205e2e2d154437675f1] [ERROR]   FixingGhostMailboxTest.oldGhostedMailboxShouldNoMoreBeAccessible » ParameterResolution
[7aa172e6365d0379fefbc205e2e2d154437675f1] [ERROR]   FixingGhostMailboxTest.webadminCanMergeTwoMailboxes » ParameterResolution No P...
[7aa172e6365d0379fefbc205e2e2d154437675f1] [ERROR]   FixingGhostMailboxTest.webadminCanMergeTwoMailboxesRights » ParameterResolution
```

File: server/container/guice/cassandra-guice/src/test/java/org/apache/james/CassandraExtension.java [hunk 1]
```diff
@@ -56,4 +59,14 @@ public void pause() {
     public void unpause() {
         cassandra.unpause();
     }
+
+    @Override
+    public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+        return parameterContext.getParameter().getType() == Host.class;
```
Comment (chibenwa) [inline server/container/guice/cassandra-guice/src/test/java/org/apache/james/CassandraExtension.java]: I'm suspicious here...

What if we add the `Host` resolution capability to other backends (akka ElasticSearch?)

Let me suggest to add  a `@Cassandra` annotation that we can access via `ParameterContext` to disambiguate that!

Comment (trantienduchn) [reply server/container/guice/cassandra-guice/src/test/java/org/apache/james/CassandraExtension.java]: I made it to `CassandraHost` to be more clear

**Comment by trantienduchn** [comment]
:green_apple:

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**